logic/resolvable/ResolvableConjunction.java

```diff
@@ -62,4 +62,9 @@ public FunctionalIterator<Concludable> allConcludables() {
                 iterate(negations()).flatMap(negated -> iterate(negated.disjunction().conjunctions()))
                         .flatMap(conj -> conj.allConcludables()));
     }
+
+    @Override
+    public String toString() {
+        return "ResolvableConjunction{" +  conjunctionPattern + '}';
```

Comment:
note: we try to avoid the default `toString` - you can get rid of the `ResolvableConjunction{` part